Dockerfile

```
@@ -18,56 +18,38 @@
 FROM node:18-bookworm as build
 RUN node --version && npm --version
 
-# Prepare the folder to enable non-root, otherwise npm will refuse to run the postinstall
-RUN mkdir /vault
-RUN chown node:node /vault
-USER node
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 # Can be a tag, release, but prefer a commit hash because it's not changeable
 # https://github.com/bitwarden/clients/commit/${VAULT_VERSION}
 #
 # Using https://github.com/bitwarden/clients/releases/tag/web-v2024.5.1
 ARG VAULT_VERSION=9823f69c9d17e2d94de1cc005e01202dd95f0647
+ENV VAULT_VERSION=$VAULT_VERSION
+ENV VAULT_FOLDER=bw_clients
 
-WORKDIR /vault
-RUN git -c init.defaultBranch=main init && \
-    git remote add origin https://github.com/bitwarden/clients.git && \
-    git fetch --depth 1 origin "${VAULT_VERSION}" && \
-    git -c advice.detachedHead=false checkout FETCH_HEAD
+RUN mkdir /bw_web_builds
+WORKDIR /bw_web_builds
 
-COPY --chown=node:node patches /patches
-COPY --chown=node:node resources /resources
-COPY --chown=node:node scripts/apply_patches.sh /apply_patches.sh
+COPY patches ./patches
+COPY resources ./resources
+COPY scripts ./scripts
 
-RUN bash /apply_patches.sh
+RUN ./scripts/checkout_web_vault.sh; \
+	./scripts/patch_web_vault.sh; \
+	./scripts/build_web_vault.sh; \
+	mv $VAULT_FOLDER/apps/web/build ./web-vault
```

```suggestion
RUN ./scripts/checkout_web_vault.sh
RUN ./scripts/patch_web_vault.sh
RUN ./scripts/build_web_vault.sh
RUN mv $VAULT_FOLDER/apps/web/build ./web-vault
```

I think if you separate the statements docker can cache the intermediate results as separate layer.

---

I'm guessing when building locally in case of error ?
Otherwise, in general more layers are not exactly a plus, no ?

---

In this case it doesn't matter for the end result. We use a from scratch there. So everything before that will not end up into the final image. 

And having separate runs here is better indeed. 
Good catch.